scout/adapter/mongo/case.py

```diff
@@ -288,29 +297,30 @@ def _update_case_id_query(self, query, id_list):
 
     def cases(
         self,
-        owner=None,
-        collaborator=None,
-        query=None,
-        skip_assigned=False,
-        has_causatives=False,
-        reruns=False,
-        rerun_monitor=False,
-        finished=False,
-        research_requested=False,
-        is_research=False,
-        has_rna_data=False,
-        status=None,
-        phenotype_terms=False,
-        group=None,
-        pinned=False,
-        cohort=False,
-        name_query=None,
-        yield_query=False,
-        within_days=None,
-        assignee=None,
-        verification_pending=None,
-        has_clinvar_submission=None,
-    ):
+        owner: Optional[str] = None,
```

Comment:
🥇with the Optional you see right away what is required and what is not!

Reply:
Ah, but that is the thing - you don’t - see the pydantic manual quote. The “Optional[x]” typing term simply means “Union[x,None]” which may or may not coincide with being an optional argument in the Python function sense. Consider eg an optional argument for an integer with an integer default value. It would still not be appropriate to use Optional on it. Or conversely a required parameter that has no default but can legitimately take the None value. It is not optional but it is appropriate to typing mark it Optional. To me, it would have made more sense to name the Union[x, None] something else, say “Discontinuous” or “Punctuated”.

Reply:
“Noneable”? 😸

Reply:
> “Noneable”? 😸

Open an issue to propose it in their repo! 😆

Reply:
> Ah, but that is the thing - you don’t - see the pydantic manual quote. The “Optional[x]” typing term simply means “Union[x,None]” which may or may not coincide with being an optional argument in the Python function sense.

In fact in this code I would have expected all arguments with a default being set to Optional, but whatever!

Reply:
Right? Thats one direction it is silly. A bool with a default is not Optional; it can be False but not None. It is however optional - it is not a required argument in a function  call. The difference a capital letter makes…